engine/src/test/scala/cromwell/CromwellTestkitSpec.scala

```diff
@@ -442,33 +447,28 @@ abstract class CromwellTestkitSpec extends TestKit(new CromwellTestkitSpec.TestW
     }
   }
 
-  def getWorkflowMetadata(workflowId: WorkflowId, key: Option[String] = None)(implicit ec: ExecutionContext): JsObject = {
+  def getWorkflowMetadata(workflowId: WorkflowId, key: Option[String] = None, serviceRegistryActor: ActorRef)(implicit ec: ExecutionContext): JsObject = {
     // MetadataBuilderActor sends its response to context.parent, so we can't just use an ask to talk to it here
-    val supervisor = TestActorRef(new Actor() {
-      var originalSender = system.deadLetters
-
-      override def receive: Receive = {
-        case m: RequestComplete[_] =>
-          originalSender ! m
-        case m: GetMetadataQueryAction =>
-          originalSender = sender
-          val mdba = TestActorRef(MetadataBuilderActor.props(serviceRegistryActor), self, "mdba" )
-          mdba ! m
-      }
-    })
-
     val message = GetMetadataQueryAction(MetadataQuery(workflowId, None, key, None, None))
-    Await.result(supervisor.ask(message).mapTo[RequestComplete[(String,JsObject)]], Duration.Inf).response._2
+    val parentProbe = TestProbe()
+
+    TestActorRef(MetadataBuilderActor.props(serviceRegistryActor), parentProbe.ref, s"MetadataActor-${UUID.randomUUID()}") ! message
+    val metadata = parentProbe.expectMsgPF(TimeoutDuration) {
+      case response: RequestComplete[(StatusCode, JsObject)] @unchecked => response.response._2
```

Comment:
Just for the benefit of understanding this easily in the future , could you add a quick comment about the `@unchecked`?